### ISSUE TITLE
Mapping from Java to J2objc libraries

### ISSUE THREAD

**Issue by advayDev1** [j2objc-contrib/j2objc-gradle#180]
In order to properly do #41, we need to map .pom-style dependencies to j2objc libraries.  How do we know a particular library like junit v4.1 is actually "-ljunit" for j2objc?  How do we know RandomLibrary1 can actually have its j2objc version downloaded, for example, from our org?

This needs a design doc I think.

**Comment by brunobowden** [comment]
Agreed, this is an issue for J2ObjC itself. At the moment, it seems like the libraries will advance in lock step when updating the j2objc version. I very carefully make sure that I use the same library version in my java build as the J2ObjC build - it should warn people who are not doing that.

**Comment by advayDev1** [comment]
@brunobowden  - Not completely j2objc's issue I think.  We are the ones trying to support automatic dependencies, they don't seem to need to (cc @tomball - for reference, we want to automatically map from gradle/ant/etc. Java library dependencies to corresponding j2objc dist JARs and .a files, where possible).

Also, we want to support non-j2objc-standard libraries in the future, which again requires us to build such a mapping.
